tests/sagemaker-tests/test_sagemaker_utils.py

```diff
@@ -221,3 +233,159 @@ def test_wrap_model_artifacts():
+    ret = upload_data_to_s3('s3://a_bucket/a_path/', './model.tar.gz', 'session')
+    mock_s3uploader.assert_called_once_with('./model.tar.gz', 's3://a_bucket/a_path/',
+                                            sagemaker_session='session')
+
+    assert ret == 's3://a_bucket/a_path/model.tar.gz'
+
+    # Test case 2: mock unsucessful upload
+    ret = upload_data_to_s3('s3://a_bucket/a_path/', './model.tar.gz', 'session')
+    with pytest.raises(AssertionError):
+        mock_s3uploader.assert_called_once_with('s3://a_bucket/a_path/', './model.tar.gz',
+                                                sagemaker_session='session')
+    
+
+def local_test_check_tarfile_s3_object(input_value, expected_result):
+    """ Test the check if a model tarfile exist in the given S3 url location.
+
+    ## This test should NOT be ran in CI pipeline, but in local environment only! ##
+    
+    Testers need to provide an S3 url as the input value, and True or False as the
+    expected result in the python command line.
+
+    """
+    s3_url = input_value
+
+    if expected_result.lower() == 'true':
+        assert check_tarfile_s3_object(s3_url)
+    else:
+        assert not check_tarfile_s3_object(s3_url)
+
+
+def local_test_upload_data_to_s3(input_value, expected_result=None):
+    """ Test the check if can upload a file to the given S3 url location.
+
+    ## This test should NOT be ran in CI pipeline, but in local environment only! ##
+    
+    Testers need to provide an S3 url as the input value. This test function will create
+    a temporary emtpy file as the data to be uploaded.
+    """
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        data_path = os.path.join(tmpdirname, 'data.txt')
+        create_dummy_file(data_path)
+
+        # prepare sessions
+        b3_session = boto3.Session(region_name='us-west-2')
+        sm_session = sm.Session(boto_session=b3_session)
+
+        s3_path = input_value
+        ret = upload_data_to_s3(s3_path, data_path, sm_session)
+
+    expected_result = input_value + '/data.txt'
+    assert ret == expected_result
+
+    # delete the uploaded object
+    parsed_url = urlparse(ret)
+    bucket_name = parsed_url.netloc
+    object_key = parsed_url.path.lstrip('/')
+    s3 = boto3.client('s3')
+    s3.delete_object(Bucket=bucket_name, Key=object_key)
+
+
+if __name__ == '__main__':
+    pass
+    # Should only run this in local environment where service access is configured.
+    # Comment out the parser when run locally
+    # parser = argparse.ArgumentParser('Local test for services')
+    # parser.add_argument('--input-value', type=str, help='Input value of a test function.')
+    # parser.add_argument('--expected-result', type=str, help='The expected result of a test.')
+    # args = parser.parse_args()
+
+    # Comment out all excpet the one to be tested.
+    # local_test_check_tarfile_s3_object(args.input_value, args.expected_result)
+
+    # local_test_upload_data_to_s3(args.input_value)
```

Comment:
Generally we should not include dead code in the repo. 

If we want to test for existence of S3 files we can use [moto](https://docs.getmoto.org/en/latest/docs/getting_started.html#pytest-fixtures-example-usage) otherwise such tests should go to integration tests

Reply:
Removed both local tests. By the way, moto tests are still not real AWS service interactions. We will need some regression test for these functions.